### ISSUE TITLE
Differential weight decays

### ISSUE THREAD

**Issue by andrewargatkiny** [andrewargatkiny/dense-attention#36]
1) Currently, our framework supports null weight decay in layernorms, biases, and, optionally, in pooler and embeddings parameters. However, many architectures, such as SSMs, require no weight decay in certain other parameters. I propose to create a tooling which would allow to select specific parameter names (whether on the whole model level or on the level of individual layers  – see #35) which should have 0 weight decay. I believe this is a more configurable and transferable way than hardcoding them as boolean flags in models' or framework code.

2) Likewise, some architectures benefit from differential learning rates for specific parameters. It could be done in the same way as with weight decays –  with configs specifying parameter names and their learning rates.

At the implementation level, one of possible solutions could be to parse the respective configs at the model/ layer initialization, to mark these parameters with special lr/ wd attributes, and then to handle them in `prepare_optimizer_parameters` and `update_learning_rate` functions in `deepspeed_train.py`. Special attention should also be paid to this code: 

https://github.com/andrewargatkiny/dense-attention/blob/9242fe79d209ca4b7a40279d63f303fb34c65760/deepspeed_train.py#L184-L186

I had written it on purpose long ago to mitigate the following issue: when DeepSpeed resumes training from a checkpoint, it doesn't pick up new `wd` and `lr` values from potentially changed configs (it takes the values saved in the checkpoint instead), and the new values have to be set manually after the checkpoint has loaded. The differential weight decays and learning rates should be properly processed here, too.

---

Please work on this issue in two separate PRs: one for wd (higher priority) and, afterwards, another for learning rates (lower priority).
